metrics/metrics_collector.go

```diff
@@ -0,0 +1,160 @@
+package metrics
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type metricCollector struct {
+	counters   []prometheus.Counter
+	gauges     []prometheus.Gauge
+	histograms []prometheus.Histogram
+	summarys   []prometheus.Summary
+	interval   int
+}
+
+var (
+	promRegistry = prometheus.NewRegistry() // local Registry so we don't get Go metrics, etc.
+)
+
+func newMetricCollector() metricCollector {
+	return metricCollector{}
+}
+
+func (mc *metricCollector) updateCounter() {
+	for {
+		time.Sleep(time.Second * time.Duration(mc.interval))
```

Comment:
Perhaps store `mc.interval` as a `time.Duration` and cast it once upon initializing the `metricCollector` rather than every time it is used.